geode-tcp-server/src/test/java/org/apache/geode/distributed/internal/tcpserver/HostAndPortTest.java

```diff
@@ -62,28 +73,97 @@ public void Test_equals_LocatorAddress_from_resolved_and_unresolved_SocketAddres
     assertThat(locator1.toString()).contains("socketInetAddress");
   }
 
+  @Test
+  public void constructorWithNoHostName() {
+    HostAndPort hostAndPort = new HostAndPort(null, 8091);
+    assertThat(hostAndPort.getAddress()).isNotNull();
+    assertThat(hostAndPort.getHostName()).isEqualTo("0.0.0.0");
+    assertThat(hostAndPort.getPort()).isEqualTo(8091);
+    assertThat(hostAndPort.getSocketInetAddress()).isNotNull();
+  }
+
+  @Test
+  public void testEquality() {
+    HostAndPort hostAndPort1 = new HostAndPort("127.0.0.1", 8091);
+    HostAndPort hostAndPort2 = new HostAndPort("127.0.0.1", 8091);
+    HostAndPort hostAndPort3 = new HostAndPort("127.0.0.1", 8092);
+    assertThat(hostAndPort1.getSocketInetAddress()).isSameAs(hostAndPort1.getSocketInetAddress());
+    assertThat(hostAndPort1).isEqualTo(hostAndPort1);
+    assertThat(hostAndPort1).isEqualTo(hostAndPort2);
+    assertThat(hostAndPort1).isNotEqualTo(hostAndPort3);
+    assertThat(hostAndPort1.equals(null)).isFalse();
+  }
+
+  @Test
+  public void testSerializationWithNumericAddress() throws IOException, ClassNotFoundException {
+    DSFIDSerializer dsfidSerializer = new DSFIDSerializerFactory().create();
+    dsfidSerializer.registerDSFID(HOST_AND_PORT, HostAndPort.class);
+    HostAndPort hostAndPort1 = new HostAndPort("127.0.0.1", 8091);
+    BufferDataOutputStream out = new BufferDataOutputStream(100, Version.CURRENT);
+    dsfidSerializer.getObjectSerializer().writeObject(hostAndPort1, out);
+    HostAndPort hostAndPort2 = dsfidSerializer.getObjectDeserializer()
+        .readObject(new ByteArrayDataInput(out.toByteArray()));
+    assertThat(hostAndPort1).isEqualTo(hostAndPort2);
+    assertThat(hostAndPort2).isEqualTo(hostAndPort1);
+  }
+
+  @Test
+  public void testSerializationWithUnresolvableHostName()
+      throws IOException, ClassNotFoundException {
+    DSFIDSerializer dsfidSerializer = new DSFIDSerializerFactory().create();
+    dsfidSerializer.registerDSFID(HOST_AND_PORT, HostAndPort.class);
+    HostAndPort hostAndPort1 = new HostAndPort("unresolvable host name", 8091);
+    BufferDataOutputStream out = new BufferDataOutputStream(100, Version.CURRENT);
+    dsfidSerializer.getObjectSerializer().writeObject(hostAndPort1, out);
+    HostAndPort hostAndPort2 = dsfidSerializer.getObjectDeserializer()
+        .readObject(new ByteArrayDataInput(out.toByteArray()));
+    assertThat(hostAndPort1).isEqualTo(hostAndPort2);
+    assertThat(hostAndPort2).isEqualTo(hostAndPort1);
+    assertThat(hostAndPort1.getAddress()).isNull();
+    assertThat(hostAndPort2.getAddress()).isNull();
+    assertThat(hostAndPort2.getSocketInetAddress()).isNotNull();
+    assertThat(hostAndPort1.getSocketInetAddress().isUnresolved()).isTrue();
+    assertThat(hostAndPort2.getSocketInetAddress().isUnresolved()).isTrue();
+  }
+
+  @Test
+  public void testSerializationWithNoHostName() throws IOException, ClassNotFoundException {
```

Comment:
why would `HostAndPort` allow no hostname? is that only allowed during deserialization?

Reply:
I think it was for a local address.  That surprised me as well.

Reply:
oy ok